### PR TITLE
Feat(SRE-378): add default capacity provider

### PR DESCRIPTION
**Pull request by sxmchxn**
This includes a new ASG to be used with a capacity provider when running scheduled tasks. This is not to be used for general services and tasks, since they have a mix of placement strategies that doesn't work well with a capacity provider at the moment. All services and tasks should continue to use the `EC2` launch type.

This enables https://github.com/builtinx/infrastructure/pull/907

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -211,11 +211,85 @@ resource "aws_autoscaling_group" "container_instance" {
...
 #
 resource "aws_ecs_cluster" "container_instance" {
   name = "ecs${title(var.environment)}Cluster"
+  capacity_providers = [aws_ecs_capacity_provider.scheduled_tasks.name]
+  default_capacty_provider_strategy {
+    capacity_provider = aws_ecs_capacity_provider.scheduled_tasks.name
+  }
+}
+
+resource "aws_ecs_capacity_provider" "scheduled_tasks" {
```
Comment (LeeAMitchell) [inline main.tf:283]: I think it should be named more generically so that it makes sense for it to be used for things beyond scheduled tasks.

**Review by michaelanthonymain — APPROVED** [review]
LGTM 🍰

File: main.tf [hunk 2]
```diff
@@ -211,11 +211,85 @@ resource "aws_autoscaling_group" "container_instance" {
...
+  lifecycle {
+    create_before_destroy = true
+  }
+
+  root_block_device {
+    volume_type = "${var.root_block_device_type}"
+    volume_size = "${var.root_block_device_size}"
+  }
+
+  name_prefix          = "lc${title(var.environment)}ContainerInstanceScheduledTasks-"
```
Comment (LeeAMitchell) [inline main.tf:224]: I'd use name rather than name_prefix because we shouldn't have any name collisions. That said, this shouldn't be an issue to use name_prefix. However, I don't think putting the environment in the name is useful given that each of our AWS accounts are isolated by environment already. I'm not against putting env in names of things, but we should be consistent. Is the plan to start putting environment in names of things?

File: main.tf (hunk 1 above)
Comment (sxmchxn) [reply main.tf:283]: I don't have any plans to use this capacity provider for anything else besides scheduled tasks. There could be multiple capacity providers for a cluster, if that helps make this make more sense.

File: main.tf (hunk 2 above)
Comment (sxmchxn) [reply main.tf:224]: This was a copy-pasta. The `var.environment` here is specified as `ecs`, rather than develop/staging/production.

https://github.com/builtinx/infrastructure/blob/794e758b109e05bcb9ca90e5e7f122985cf6f151/aws/test/ecs.tf#L40

File: main.tf (hunk 1 above)
Comment (LeeAMitchell) [reply main.tf:283]: What benefit is there to multiple capacity providers? Downside is it reduces resource utilization. Example, you have capacity providers A, B, C. Capacity provider A was used to bring up an extra EC2 instance, but that EC2 instance is not fully utilized. Now a service using capacity provider B is trying to launch, but B doesn't have resources (A does, but it's not using A), so we bring up an unnecessary instance on B.

If we had something special about these instances, for example, we used a different instance type or size that was very different I could see some benefits. Like if you wanted to use GPU instances or had some odd tasks that required a huge amount of resources, so you only want to bring up say m5.16xlarge just for those specific tasks.

This PR sets the "scheduled tasks" capacity provider was set as the default capacity provider for the cluster. The default should have a generic name.

So in summary, we should have a generic default capacity provider and it should be named as such. If we have a need for special capacity providers, then we could create them as needed. I don't see a need and this is the default, so I think it should have a generic name (or not be defined as the default).

**Review by LeeAMitchell — APPROVED** [review]
Approving because I don't see this PR causing any technical issues.

File: main.tf (hunk 1 above)
Comment (sxmchxn) [reply main.tf:283]: Having this be the "default" provider is the easiest way for us to get scheduled tasks on a different instance type. This is probably incompatible with a more general use of capacity providers, so I would expect more ~extending~ extensive changes if we are to use capacity providers for more than scheduled tasks.
